### PR TITLE
Add documentation for CLI interaction with registry having self-signed CA

### PR DESCRIPTION
**Pull request by prkalle**
### What this PR does / why we need it
This PR adds documentation for CLI interaction with registry which has self-signed CA certificate or expired certificate
### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note
N/A
```

### CODE REVIEW

File: docs/full/README.md [hunk 1]
```diff
@@ -206,3 +206,33 @@ Signature verification could fail in the scenarios below:
    that signature verification is skipped for the repository. Users can choose to
    suppress this warning by setting the environment variable `TANZU_CLI_SUPPRESS_SKIP_SIGNATURE_VERIFICATION_WARNING`
    to `true`.
+
+## Interacting with Central repository hosted on a registry with self-signed CA or with expired CA
```
Comment (marckhouzam) [inline docs/full/README.md:210]:
```suggestion
## Interacting with a central repository hosted on a registry with self-signed CA or with expired CA
```

File: docs/full/README.md [hunk 2]
```diff
@@ -206,3 +206,33 @@ Signature verification could fail in the scenarios below:
...
+
+If user has configured central repository on custom registry (e.g. air-gaped environment) with self-signed CA or if the
```
Comment (marckhouzam) [inline docs/full/README.md:212]:
```suggestion
If a user has configured a central repository on a custom registry (e.g. air-gaped environment) with a self-signed CA or if the
```

File: docs/full/README.md [hunk 3]
```diff
@@ -206,3 +206,33 @@ Signature verification could fail in the scenarios below:
...
+registry CA
+certificate is expired, user can use the `tanzu config cert` command to configure the certificate for the registry host.
```
Comment (marckhouzam) [inline docs/full/README.md:214]:
```suggestion
certificate is expired, the user can execute the `tanzu config cert` family of commands to configure the certificate for the registry host.
```

File: docs/full/README.md [hunk 4]
```diff
@@ -206,3 +206,33 @@ Signature verification could fail in the scenarios below:
...
+
+```shell
+
+    # If the registry host is self-signed add CA certificate for the registry
+    tanzu config cert add --host test.registry.com --ca-certificate path/to/ca/ert
```
Comment (marckhouzam) [inline docs/full/README.md:219]:
```suggestion
    tanzu config cert add --host test.registry.com --ca-certificate path/to/ca/cert
```

File: docs/full/README.md [hunk 5]
```diff
@@ -206,3 +206,33 @@ Signature verification could fail in the scenarios below:
...
+
+    # If the registry is self-signed and is serving on non-default port add CA certificate for the registry
+    tanzu config cert add --host test.registry.com:8443 --ca-certificate path/to/ca/ert
```
Comment (marckhouzam) [inline docs/full/README.md:222]:
```suggestion
    tanzu config cert add --host test.registry.com:8443 --ca-certificate path/to/ca/cert
```

File: docs/full/README.md [hunk 6]
```diff
@@ -206,3 +206,33 @@ Signature verification could fail in the scenarios below:
...
+    tanzu config cert add --host test.registry.com  --insecure true
+
+```
+
+CLI uses the certificate configuration added for the registry host (using `tanzu config cert add` command ) while
+interacting with the registry.
+
+Users can update or delete the certificate configuration using the `tanzu config cert update`
+and `tanzu config cert delete` commands.
+Also, users can list down the certificate configuration using `tanzu config cert list` command.
```
Comment (marckhouzam) [inline docs/full/README.md:238]:
```suggestion
Also, users can list the certificate configuration using the `tanzu config cert list` command.
```

**Review by marckhouzam — COMMENTED** [review]
Just grammatical comments.

Do we want to mention how to deal with a proxy, or do we wait until the PR that improves that solution?

File: docs/full/README.md (hunk 1 above)
Comment (prkalle) [reply docs/full/README.md:210]: Updated. Thanks

File: docs/full/README.md (hunk 2 above)
Comment (prkalle) [reply docs/full/README.md:212]: updated

File: docs/full/README.md (hunk 3 above)
Comment (prkalle) [reply docs/full/README.md:214]: updated

File: docs/full/README.md (hunk 5 above)
Comment (prkalle) [reply docs/full/README.md:222]: updated

File: docs/full/README.md [hunk 7]
```diff
@@ -206,3 +206,33 @@ Signature verification could fail in the scenarios below:
...
+    # If the registry is self-signed or CA cert is expired, add cert configuration for the registry host with
+    # skip-cert-verify option
+    tanzu config cert add --host test.registry.com  --skip-cert-verify true
+
+    # Set to allow insecure (http) connection while interacting with host
+    tanzu config cert add --host test.registry.com  --insecure true
+
+```
+
+CLI uses the certificate configuration added for the registry host (using `tanzu config cert add` command ) while
```
[the remark this answers is not in the document]
Comment (prkalle) [reply docs/full/README.md:233]: updated

File: docs/full/README.md (hunk 6 above)
Comment (prkalle) [reply docs/full/README.md:238]: updated

**Comment by prkalle** [comment]
> Do we want to mention how to deal with a proxy, or do we wait until the PR that improves that solution?

Yes, we will update the documention for proxy certificate configuration when we have the PR to improve that solution

**Review by marckhouzam — APPROVED** [review]
LGTM

**Pull request merged**